src/base/queue_task.h

```diff
@@ -213,9 +213,8 @@ class WorkQueue {
     }
 
     ~WorkQueue() {
+        tbb::mutex::scoped_lock lock(mutex_);
```

Comment:
@michal-clapinski I think I will need some explanation here, probably I am missing the point

Reply:
Yes, definitely. See this https://github.com/codilime/contrail-controller/pull/53. I'm actually reverting a change I made before. I thought this previous change would fix error in `queue_task_test.cc` but it didn't. I'm hoping adding `WaitForIdle` will fix it once and for all.

Reply:
@michal-clapinski Ok now I see the point. However this lonely mutex seems to be a bit weird.

Reply:
Also the comment `// Shutdown() needs to be called before deleting` and then not calling it is a bit weird but I got used to it.